### PR TITLE
[IMP] web_editor: improve code style

### PR DESCRIPTION
**Pull request by xlu-odoo**
The previous styling of code (inline & block) in the web editor was lacking, which affected both its readability and visual attractiveness.

This small update tackles these issues by enhancing the color scheme andoverall design for both inline code and code block, resulting in a more legible and visually appealing presentation.

task-3620735

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by mano-odoo — APPROVED** [review]
Thank you for your work! It's way better than what we had, LGTM.

Just a minor, in convention guidelines, padding comes after the border utility class but it's not a big deal we use both order everywhere in the code.

Just wondering if we could also apply the change to the `code` text style in adddition to the inline-code applied with backticks  \`\`, I'm unsure why it behaves differently but it is the case in current master as well so maybe it's expected? @bouvyd .

**Comment by bouvyd** [comment]
> Just wondering if we could also apply the change to the code text style in adddition to the inline-code applied with backticks

ok for background, border, rounding but *not* the font-color - font color should be black, or it should be syntax-colored (Which we do not do as of now), but not colored in any way otherwise.

**Review by mano-odoo — COMMENTED** [review]
Thanks for your work, It looks good! 

Care that by applying the classes in the `OdooEditor.js` your inline-code option changed the visual result in the front-end as well. While it looks way better we should probably check with website/web editor team if that's what they want. It would be a bit weird to implement only the inline code in the frontend & backend and leave the block code styled only for the backend.

If we want the block code to be in both backend and frontend we could probably use utility classes directly on the pre element in the OdooEditor.js as you did for the inline-code.

Else if we don't want to impact the front-end at all we should apply the inline-code styling in the web_editor.backend.scss for both, rather than applying the utilities classes on the element itself.

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 1]
```diff
@@ -99,6 +99,14 @@ html, body {
     height: 100%;
 }
 
+pre {
+    border: $border-width solid $border-color;
```
Comment (mano-odoo) [inline addons/web_editor/static/src/scss/web_editor.common.scss:103]: Mind that the border-color here in frontend is using `var(--o-border-color)` which is equal to: `color-mix(in srgb, currentColor 15%, transparent)`

So if you want a consistent $o-gray-300 between inline-code and codeblock you might want to set the `--o-border-color` here as well. 👼

**Review by mano-odoo — CHANGES_REQUESTED** [review]
Thanks for your work 💪 
Just a two comments below

**Comment by xlu-odoo** [comment]
Thanks @mano-odoo for the inputs! Updated it. Can you check again?

**Review by mano-odoo — APPROVED** [review]
lgtm! Do not forget to ping sri for final review 💪

**Comment by xlu-odoo** [comment]
Thanks @mano-odoo 
@stefanorigano , can you look into this PR?

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @xlu-odoo , thanks for your work, it's a great improvement 👍 

Some remarks/questions for you in the code...
I assume that the expected visual result changed during the implementation, could you update the PR's screenshot accordingly?

Thanks!

File: addons/web_editor/static/src/scss/web_editor.frontend.scss [hunk 2]
```diff
@@ -114,3 +114,12 @@ body.editor_enable:not(.o_basic_theme) .odoo-editor-editable {
         background-color: transparent !important;
     }
 }
+
+// Adjust the color scheme for inline code elements. This is particularly 
+// necessary because the 'bg-100' class in frontend has a text color 
+// attached to the background color.
+.o_inline_code {
```
[the remark this answers is not in the document]
Comment (xlu-odoo) [reply addons/web_editor/static/src/scss/web_editor.frontend.scss:121]: True but we do apply a border via `o_inline_code p-1 border rounded bg-100 lh-lg` 

.border class using o-border-color variable, in front-end it equals to color-mix(in srgb, currentColor 15%, transparent)

Which would do:
image

Or do you mean something else? ( That it shouldn't be targeting o_inline_code? )

File: addons/web_editor/static/src/scss/web_editor.frontend.scss [hunk 3]
```diff
@@ -114,3 +114,12 @@ body.editor_enable:not(.o_basic_theme) .odoo-editor-editable {
...
+    --o-border-color: #{$o-gray-300};
+    
+    color: $o-main-code-color;
```
[the remark this answers is not in the document]
Comment (xlu-odoo) [reply addons/web_editor/static/src/scss/web_editor.frontend.scss:124]: Not sure what to use else that would be correct

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 4]
```diff
@@ -990,3 +998,11 @@ section, .oe_img_bg, [data-oe-shape-data] {
         margin-bottom: 1rem;
     }
 }
+
+code.o_inline_code {
+    padding: map-get($spacers, 1);
+    border: $border-width solid $border-color;
+    border-radius: $border-radius;
+    background-color: $gray-100;
+    line-height: 2;
```
Comment (stefanorigano) [inline addons/web_editor/static/src/scss/web_editor.common.scss:1007]: I think that this is missing a `color: $gray-900;`  😉

----
image

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 5]
```diff
@@ -990,3 +998,11 @@ section, .oe_img_bg, [data-oe-shape-data] {
         margin-bottom: 1rem;
     }
 }
+
+code.o_inline_code {
```
Comment (stefanorigano) [inline addons/web_editor/static/src/scss/web_editor.common.scss:1002]: Not a big fan of the final result. What worries me the most is the the fact that this design makes text lines containing `code` to be have an huge vertical gap and to be consequentially rendered as "independent" from the others, almost like if they belongs to a paragraph ad-hoc (see picture, first paragraph). This message is indeed a bit too verbose just to prove that adding a `code` string here in github, doesn't  affect the paragraph line-height.. I think that we need something similar.

image

To achieve the example in the second paragraph, I've briefly did this in my browser

```scss
    font-size: 85%;
    line-height: inherit;
    border: 0;
    padding: 0.2em 0.4em;
    background: var(--200);
    color: var(--900);
  ```
  
  _note_: this suggestion is `code.o_inline_code`.
  `pre` can have default font-size, border etc..

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 6]
```diff
@@ -990,3 +998,11 @@ section, .oe_img_bg, [data-oe-shape-data] {
...
+    padding: map-get($spacers, 1);
+    border: $border-width solid $border-color;
+    border-radius: $border-radius;
+    background-color: $gray-100;
```
Comment (stefanorigano) [inline addons/web_editor/static/src/scss/web_editor.common.scss:1006]: If we remove the border, `$gray-200` seems a better fit

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 7]
```diff
@@ -990,3 +998,12 @@ section, .oe_img_bg, [data-oe-shape-data] {
         margin-bottom: 1rem;
     }
 }
+
+code.o_inline_code {
+    padding: .2em .4em;
+    border-radius: $border-radius;
+    border: 0;
+    background-color: $gray-200;
```
Comment (stefanorigano) [inline addons/web_editor/static/src/scss/web_editor.common.scss:1006]: Uff.. in dark-mode it matches the fromView background-color :/
| bright | dark |
|--------|--------|
| image | image | 

I'll let you investigate but seems a good scenario for CSS variables, something like:

```scss
background-color: var(--o-inline-code-background-color, #{$gray-200});
```

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/scss/web_editor.common.scss (hunk 7 above)
Comment (xlu-odoo) [reply addons/web_editor/static/src/scss/web_editor.common.scss]: Thanks for the feedback. Updated to: 

| bright | dark |
|--------|--------|
| image | image |

**Comment by xlu-odoo** [comment]
@stefanorigano (small reminder)

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 8]
```diff
@@ -962,3 +970,12 @@ section, .oe_img_bg, [data-oe-shape-data] {
         margin-bottom: 1rem;
     }
 }
+
+code.o_inline_code {
+    padding: .2em .4em;
+    border-radius: $border-radius;
+    border: 0;
```
Comment (mano-odoo) [inline addons/web_editor/static/src/scss/web_editor.common.scss:977]: Is `border: 0;` still necessary in some context? In knowledge and website it doesn't seem to have a border by default

**Review by mano-odoo — CHANGES_REQUESTED** [review]
Aside

**Review by mano-odoo — APPROVED** [review]
LGTM

**Comment by xlu-odoo** [comment]
Hello @odoo/rd-website / @odoo/rd-framework-editor  👋 
Could you review this PR and eventually merge please?

**Review by sebgeelen — APPROVED** [review]
LGTM

**Comment by sebgeelen** [comment]
@robodoo r+

**Pull request closed without merging**